libs/barista-components/filter-field/src/filter-field.ts

```diff
@@ -805,6 +825,222 @@ export class DtFilterField<T = any>
     }
   }
 
+  /** @internal Handles keyup on the input */
+  _handleInputKeyUp(event: KeyboardEvent) {
+    const keyCode = _readKeyCode(event);
+
+    if (
+      keyCode === LEFT_ARROW &&
+      this._inputEl.nativeElement.selectionStart === 0
+    ) {
+      // The input element moves the cursor on keydown but the navigation with arrow keys is on keyup,
+      // causing the cursor to jump over position 0 in the input and focuses the first focusable tag element.
+      // For that, allowArrowKeyFocusOnTags retains getting to position 0 before focusing a tag element.
+      if (this.allowArrowKeyFocusOnTags || !this._inputValue.length) {
+        this._handleArrowKeyNavigation({ left: true });
+      } else {
+        this.allowArrowKeyFocusOnTags = true;
+      }
+    } else if (
+      keyCode === RIGHT_ARROW &&
+      this._inputEl.nativeElement.selectionStart === this._inputValue.length
+    ) {
+      // The input element moves the cursor on keydown but the navigation with arrow keys is on keyup,
+      // causing the cursor to jump over the last position in the input and focuses the first focusable tag element.
+      // For that, allowArrowKeyFocusOnTags retains getting the cursor to the last position before focusing a tag element.
+      if (this.allowArrowKeyFocusOnTags || !this._inputValue.length) {
+        this._handleArrowKeyNavigation({ right: true });
+      } else {
+        this.allowArrowKeyFocusOnTags = true;
+      }
+    }
+  }
+
+  /** @internal Handles the navigation through the list of tags when using arrow keys */
+  _handleArrowKeyNavigation(navigationEvent: {
+    /** Contains the currently focused tag from which the event was fired from */
+    tag?: DtFilterFieldTag;
+    /** Whether to step back */
+    left?: boolean;
```

Comment:
```suggestion
    direction: 'left' | 'right';
```
Again here - right and left can never be true at the same time, use one variable instead - makes it easier